OSDOCS:14528 Docs for On Cluster Layering and a seamless SNO experience/Disconnected Support (#92844)
openshift/openshift-docs: modules/coreos-layering-configuring-on.adoc

You should not need to interact with these new objects or the `machine-os-builder` pod. However, you can use all of these resources for troubleshooting, if necessary.

You need a separate `MachineOSConfig` CR for each machine config pool where you want to use a custom layered image. You can apply a custom layered image by using any machine config pool in your cluster, including the control plane, worker, or custom pool.

[NOTE]
====

> For single node OpenShift (SNO) clusters, you can use only the control plane machine config pool to apply a custom layered image.

Author: @cheesesashimi Is this all we need to cover this issue? It appears from [your comments](OCL work as-is on SNO) that "OCL work as-is on SNO" save for this one caveat.

You need a separate `MachineOSConfig` CR for each machine config pool where you want to use a custom layered image.

[NOTE]
====

> For {sno} clusters, you can apply a custom layered image to the control plane node only. There should only be one `MachineOSConfig` object, which corresponds to control plane machine config pool.

Reviewer: ```suggestion
For {sno} clusters, you can apply a custom layered image to the control plane node only. Only one `MachineOSConfig` object must exist, which corresponds to the control plane machine config pool.
```

Reviewer: Yes, this is much better for an optional component. Thanks for asking!
